### PR TITLE
Webpack config improvements

### PR DESCRIPTION
**Pull request by xcv58**
- Use CopyWebpackPlugin to generate 'manifest.json` file dynamically
- Use CleanWebpackPlugin to clear build folder
- Remove `utils/prepare.js` and `utils/generate_manifest.js` files

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -2,6 +2,8 @@ var webpack = require("webpack"),
     path = require("path"),
     fileSystem = require("fs"),
     env = require("./utils/env"),
+    CleanWebpackPlugin = require('clean-webpack-plugin'),
+    CopyWebpackPlugin = require('copy-webpack-plugin'),
```
Comment (samuelsimoes) [inline webpack.config.js]: Double quotes please to keep the code style.

File: webpack.config.js [hunk 2]
```diff
@@ -49,10 +51,23 @@ var options = {
     alias: alias
   },
   plugins: [
+    // clean de dist folder
```
Comment (samuelsimoes) [inline webpack.config.js]: "clean **the** dist folder" right?

File: webpack.config.js [hunk 3]
```diff
@@ -49,10 +51,23 @@ var options = {
...
+    new CleanWebpackPlugin(["build"]),
     // expose and write the allowed env vars on the compiled bundle
     new webpack.DefinePlugin({
       "process.env.NODE_ENV": JSON.stringify(env.NODE_ENV)
     }),
+    new CopyWebpackPlugin([{
+      from: 'src/manifest.json',
```
Comment (samuelsimoes) [inline webpack.config.js]: Double quotes, please.

**Comment by xcv58** [comment]
@samuelsimoes thanks, fixed.

**Comment by samuelsimoes** [comment]
@xcv58 can you squash the fix commit into the previous commit?

**Comment by xcv58** [comment]
You can squash just by a click: https://help.github.com/articles/merging-a-pull-request/#merging-a-pull-request-on-github

File: webpack.config.js [hunk 4]
```diff
@@ -49,10 +51,23 @@ var options = {
...
       "process.env.NODE_ENV": JSON.stringify(env.NODE_ENV)
     }),
+    new CopyWebpackPlugin([{
+      from: "src/manifest.json",
+      transform: function (content, path) {
+        // generates the manifest file using the package.json informations
+        return Buffer.from(JSON.stringify({
+          description: process.env.npm_package_description,
+          version: process.env.npm_package_version,
+          ...JSON.parse(content.toString())
```
Comment (mike-casas) [inline webpack.config.js]: for this feature is necessary merge this PR  #27

**Comment by xcv58** [comment]
Thanks @samuelsimoes

**Pull request merged**